zero/src/rpc/zeth.rs

```diff
@@ -0,0 +1,35 @@
+use alloy::{providers::Provider, rpc::types::eth::BlockId, transports::Transport};
+use trace_decoder::BlockTrace;
+
+use super::fetch_other_block_data;
+use crate::prover::BlockProverInput;
+use crate::provider::CachedProvider;
+
+pub async fn block_prover_input<ProviderT, TransportT>(
+    cached_provider: std::sync::Arc<CachedProvider<ProviderT, TransportT>>,
+    target_block_id: BlockId,
+    checkpoint_block_number: u64,
+) -> anyhow::Result<BlockProverInput>
+where
+    ProviderT: Provider<TransportT>,
+    TransportT: Transport + Clone,
+{
+    let block_number = match target_block_id {
+        BlockId::Number(block_number) => block_number,
+        _ => return Err(anyhow::anyhow!("block number expected")),
```

Comment:
We handle case where block hash is provided in the `CLI` for other blockchain clients, could we do the same for zeth? In the worst case that `zeth` api does not support hashes, we could get the block by hash and learn the block number.

Reply:
https://github.com/0xPolygonZero/zk_evm/issues/807